src/Agent.Worker/Build/GitCommandManager.cs

```diff
@@ -83,6 +81,15 @@ public class GitCommandManager : AgentService, IGitCommandManager
             }
 
             context.Debug($"Find git from agent's external directory: {_gitPath}.");
+            context.Debug($"Prepend git directory '{Path.GetDirectoryName(_gitPath)}' to PATH env.");
+
+#if OS_WINDOWS
+            string path = Environment.GetEnvironmentVariable("Path");
+            Environment.SetEnvironmentVariable("Path", $"{Path.GetDirectoryName(_gitPath)}{Path.PathSeparator}{path}");
+#else
+            string path = Environment.GetEnvironmentVariable("PATH");
+            Environment.SetEnvironmentVariable("PATH", $"{Path.GetDirectoryName(_gitPath)}{Path.PathSeparator}{path}");
```

Comment:
It would be better to use string builder and only append the separator if `PATH` is not empty. On OSX/Linux, leading `;`, trailing `;`, and `;;` in the `PATH` indicates that the working directory should be searched during command resolution - and is considered a security risk.